### PR TITLE
Refactor watch for DRYness

### PR DESCRIPTION
**Pull request by bcrochet**
The "watch for resource" functionality was duplicated. This extracts the function,
and no longer repeats itself.

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

File: certification/internal/policy/operator/deployable_by_olm.go [hunk 1]
```diff
@@ -234,91 +234,106 @@ func (p *DeployableByOlmCheck) grantRegistryPermissionToServiceAccount(ctx conte
 	return nil
 }
 
+type watchFunc func(ctx context.Context, engine cli.OpenshiftEngine, name, namespace string) (string, bool, error)
+
+func watch(ctx context.Context, engine cli.OpenshiftEngine, name, namespace string, timeout time.Duration, doneChannel chan string, timeoutChannel chan error, fn watchFunc) {
+	ctx, cancel := context.WithTimeout(ctx, subscriptionTimeout)
```
Comment (samira-barouti) [inline certification/internal/policy/operator/deployable_by_olm.go:240]: I feel the second arg should be the `timeout` arg.

File: certification/internal/policy/operator/deployable_by_olm.go [hunk 2]
```diff
@@ -234,91 +234,106 @@ func (p *DeployableByOlmCheck) grantRegistryPermissionToServiceAccount(ctx conte
...
+	defer cancel()
+	for {
+		log.Debug("Waiting for object to become ready...")
```
Comment (samira-barouti) [inline certification/internal/policy/operator/deployable_by_olm.go:243]: Can we add a bit more details on what is the object we are trying to fetch?

File: certification/internal/policy/operator/deployable_by_olm.go [hunk 3]
```diff
@@ -234,91 +234,106 @@ func (p *DeployableByOlmCheck) grantRegistryPermissionToServiceAccount(ctx conte
 	return nil
 }
 
+type watchFunc func(ctx context.Context, engine cli.OpenshiftEngine, name, namespace string) (string, bool, error)
```
Comment (samira-barouti) [inline certification/internal/policy/operator/deployable_by_olm.go:237]: Are we using the `timeout time.Duration`? Back to the comment that I left earlier, I think line 240 should be `ctx, cancel := context.WithTimeout(ctx, timeout)`.

Comment (samira-barouti) [reply certification/internal/policy/operator/deployable_by_olm.go:237]: Uh I meant to tag line 239. My bad :(

Comment (bcrochet) [reply certification/internal/policy/operator/deployable_by_olm.go:237]: Oh... THAT'S what you were saying. I thought you were talking about the actual function order. Not the WithTimeout call... Let me look at that.

Comment (samira-barouti) [reply certification/internal/policy/operator/deployable_by_olm.go:237]: I have to do a better job at explaining my point. Thank you for the fix!

**Review by samira-barouti — APPROVED** [review]
/lgtm

**Pull request merged**